src/hub/Hub.sol

```diff
@@ -337,30 +337,49 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler {
     }
 
     /// @inheritdoc IHub
-    function createHolding(ShareClassId scId, AssetId assetId, IERC7726 valuation, bool isLiability, uint24 prefix)
+    function createHolding(ShareClassId scId, AssetId assetId, IERC7726 valuation, AccountId assetAccount, AccountId equityAccount, AccountId lossAccount, AccountId gainAccount) 
```

Comment:
Not sure what's the reason you removed the kind from the account. That adds a new level to the mapping storages

If you pass an array of `uint24`, you could have the same behavior as before adding the `kind`

Reply:
The array also would help with the contract size issue in the Hub